glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -598,23 +594,19 @@ open class GleanInternalAPI internal constructor () {
      */
     @Synchronized
     internal fun registerPingType(pingType: PingType) {
-        if (!this.isInitialized()) {
-            pingTypeQueue.add(pingType)
-        } else {
+        if (this.isInitialized()) {
             LibGleanFFI.INSTANCE.glean_register_ping_type(
                 handle,
                 pingType.handle
             )
-
-            // In test mode we still need to keep track of pings, so they get re-registered after a reset.
-            // This state is kept across Glean resets (which only happen in test mode).
-            // This state is cleared in production-mode in `Glean.initialize`,
-            // as the ping objects themselves will live forever and Glean can't be reset.
-            @Suppress("EXPERIMENTAL_API_USAGE")
-            if (Dispatchers.API.testingMode) {
-                pingTypeQueue.add(pingType)
-            }
         }
+
+        // We need to keep track of pings, so they get re-registered after a reset.
+        // This state is kept across Glean resets, which should only ever happen in test mode.
+        // Or by the instrumentation tests (`connectedAndroidTest`), which relaunches the application activity,
+        // but not the whole process, meaning globals, such as the ping types, still exist from the old run.
+        // It's a set and keeping them around forver should not have much of an impact.
```

Comment:
Would it make sense to try to reset in the `GleanTestRunner`?

Reply:
Yeah, I'm too am curious if we can fix this in a testing-only way...

Reply:
I considered it, but decided against it for reasons:

* Low overhead even in production
* It's yet another configuration/state the GleanTestRunner runs in compared to unit tests and production mode.
* I'd like the tests to run as close to production as possible, otherwise we might miss other problems
* It was easier to remove code than to add :P